lib/shopwareStoreCommander.js

```diff
@@ -0,0 +1,364 @@
+    }
+
+    /**
+     * @param {Object} plugin
+     * @param {Array} fields
+     * @return {Promise}
+     */
+    loadExtraPluginFields(plugin, fields) {
+        plugin.scsLoadedExtraFields = plugin.scsLoadedExtraFields || [];
+        // Load all extra fields
+        const extraFieldPromises = fields.map(field => this.client.get(plugin[field]).then((res) => {
+            plugin[field] = res.data;
+            // Mark the extra field as loaded
+            if (plugin.scsLoadedExtraFields.indexOf(field) === -1) {
+                plugin.scsLoadedExtraFields.push(field);
+            }
+        }));
+
+        return Promise.all(extraFieldPromises).then(() => plugin);
+    }
+
+    /**
+     * Apply the given data to the plugin, without overwriting any extra fields that were
+     * previously loaded and stored in the plugin using 'loadExtraPluginFields()'.
+     *
+     * @param {Object} plugin
+     * @param {Object} data
+     */
+    updatePluginData(plugin, data) {
+        Object.keys(data).forEach((key) => {
+            // Only overwrite the field, if it is not a loaded extra field
+            if (plugin[key] && plugin.scsLoadedExtraFields && plugin.scsLoadedExtraFields.indexOf(key) === -1) {
+                plugin[key] = data[key];
+            }
+        });
+    }
+
+    /**
+     * @param {Object} plugin
+     * @param {Object} lastSale to compare against for db sync - optional
+     * @return {Promise}
+     */
+    async getSalesForPlugin(plugin, lastSale) {
+        const limit = 500;
+        const sales = [];
+        let currOffset = 0;
+        let tmpArr = [];
+        let exit = false;
+        do {
+            // eslint-disable-next-line no-await-in-loop
+            tmpArr = await this.client.get('pluginsales/list', {
+                params: {
+                    offset: currOffset,
+                    limit,
+                    producerId: this.accountData.producerId,
+                    pluginId: plugin.id,
+                },
+            });
+            tmpArr = tmpArr.data;
+            for (let i = 0; i < tmpArr.length; i += 1) {
+                const sale = tmpArr[i];
+                if (lastSale &&
+                    sale.creationDate === lastSale.bookingDate &&
+                    (sale.priceModel.price ? sale.priceModel.price : 0) === lastSale.amount &&
+                    sale.licenseShop.domain === lastSale.domain &&
+                    sale.priceModel.type === lastSale.type) {
+                    exit = true;
+                    break;
+                } else {
+                    sales.push(sale);
+                }
+            }
+            currOffset += limit;
+        } while (tmpArr.length !== 0 && !exit);
+
+        return sales;
+    }
+
+    /**
+     * @param {Object} plugin
+     * @param {Object} lastBrokerage to compare against for db sync - optional
+     * @return {Promise}
+     */
+    async getBrokeragesForPlugin(plugin, lastBrokerage) {
```

Comment:
The correct english term here is "commission". I think we should call it that and maybe add a side note that Shopware calls it a "brokerage".